sqlglot/optimizer/merge_subqueries.py

```diff
@@ -176,6 +176,7 @@ def _outer_select_joins_on_inner_select_join():
 
     return (
         isinstance(outer_scope.expression, exp.Select)
+        and not outer_scope.expression.is_star
```

Comment:
This was motivated by the same example as in `eliminate_subqueries`, where the schema is unknown:

```python
>>> from sqlglot.optimizer import optimize
>>> optimize("SELECT * FROM ((SELECT c FROM t1) JOIN t2)").sql()
'SELECT * FROM ("t1" AS "t1" CROSS JOIN "t2" AS "t2")'
```

Notice how the `c` projection was dropped in the optimized form.